File: 3. semester projekt/pc_store/Exceptions/ProductNotExistException.cs
using System.Runtime.Serialization;

namespace Exceptions
{
    [DataContract]
    public class ProductNotExistException
    {
        [DataMember]
        public string Message { get; set; }
        /// <summary>
        /// This exception tells if the product does not exist in the system. 
        /// </summary>
        /// <param name="product"></param>
        /// The product that does not exist in the system. 
        public ProductNotExistException(string product)
        {
            Message = $"Varenr. {product} eksisterer ikke i systemet";
        }
        
    }
}

File: 1. semester projekt/1Sem_Projekt_4_BlueJ/exceptions/ProductNotExistException.java
package exceptions;

public class ProductNotExistException extends Exception
{
    int barcode;
    
    public ProductNotExistException(int barcode)
    {
        this.barcode = barcode;
    }

    public int getBarcode()
    {
        return barcode;
    }
    
    public String toString()
    {
        return "Produktet med stregkoden " + getBarcode() + " eksisterer ikke";
    }

}
